crates/sdk/src/masp.rs

```diff
@@ -35,107 +35,112 @@ use crate::{token, MaybeSend, MaybeSync};
 #[allow(clippy::result_large_err)]
 fn extract_masp_tx(
     tx: &Tx,
-    masp_refs: &MaspTxRefs,
-) -> Result<Vec<Transaction>, Error> {
+    masp_ref: &MaspTxRef,
+) -> Result<Transaction, Error> {
     // NOTE: It is possible to have two identical references in a same batch:
     // this is because, some types of MASP data packet can be correctly executed
     // more than once (output descriptions). We have to make sure we account for
     // this by using collections that allow for duplicates (both in the args
     // and in the returned type): if the same reference shows up multiple
     // times in the input we must process it the same number of times to
     // ensure we contruct the correct state
```

Comment:
this comment doesn't make sense anymore, we're returning a single `Transaction` (rather than a `Vec` of them)